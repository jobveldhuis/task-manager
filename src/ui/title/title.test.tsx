import { render, screen } from "@testing-library/react-native";
import { Title, variants } from "./index";

describe("Title", () => {
  it("should match snapshots for each variant without icon", async () => {
    const snapshots = variants.map((variant) => {
      const component = (
        <Title variant={variant}>Lorem ipsum dolor sit amet</Title>
      );
      return expect(render(component).toJSON()).toMatchSnapshot();
    });

    await Promise.all(snapshots);
  });

  it("should match snapshots for each variant with icon", async () => {
    const snapshots = variants.map((variant) => {
      const component = (
        <Title variant={variant} icon="calendar-alt">
          Lorem ipsum dolor sit amet
        </Title>
      );
      return expect(render(component).toJSON()).toMatchSnapshot();
    });

    await Promise.all(snapshots);
  });

  it("should render without icon when omitted from props", () => {
    render(<Title>This should not have an icon</Title>);
    expect(screen.getByText("This should not have an icon")).toBeTruthy();

    // This container only exists when we do render an icon.
    expect(screen.queryByTestId("title-container")).toBeFalsy();
  });

  it("should render with icon when specified in props", () => {
    render(<Title icon="calendar-alt">This should have an icon</Title>);

    // This container only exists when we do render an icon.
    expect(screen.getByTestId("title-container")).toBeTruthy();
  });
});
