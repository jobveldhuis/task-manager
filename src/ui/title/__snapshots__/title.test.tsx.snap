// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should match snapshots for each variant with icon 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 0,
        "flexDirection": "row",
        "gap": 16,
      },
      undefined,
    ]
  }
  testID="title-container"
>
  <Text />
  <Text
    style={
      {
        "color": "#000C49",
        "fontSize": 24,
        "fontWeight": "bold",
      }
    }
  >
    Lorem ipsum dolor sit amet
  </Text>
</View>
`;

exports[`Title should match snapshots for each variant with icon 2`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 0,
        "flexDirection": "row",
        "gap": 16,
      },
      undefined,
    ]
  }
  testID="title-container"
>
  <Text />
  <Text
    style={
      {
        "color": "#000",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Lorem ipsum dolor sit amet
  </Text>
</View>
`;

exports[`Title should match snapshots for each variant with icon 3`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 0,
        "flexDirection": "row",
        "gap": 16,
      },
      undefined,
    ]
  }
  testID="title-container"
>
  <Text />
  <Text
    style={
      {
        "color": "#000",
        "fontSize": 14,
        "fontWeight": "bold",
      }
    }
  >
    Lorem ipsum dolor sit amet
  </Text>
</View>
`;

exports[`Title should match snapshots for each variant without icon 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 0,
        "flexDirection": "row",
        "gap": 16,
      },
      undefined,
    ]
  }
  testID="title-container"
>
  <Text
    style={
      {
        "color": "#000C49",
        "fontSize": 24,
        "fontWeight": "bold",
      }
    }
  >
    Lorem ipsum dolor sit amet
  </Text>
</View>
`;

exports[`Title should match snapshots for each variant without icon 2`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 0,
        "flexDirection": "row",
        "gap": 16,
      },
      undefined,
    ]
  }
  testID="title-container"
>
  <Text
    style={
      {
        "color": "#000",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Lorem ipsum dolor sit amet
  </Text>
</View>
`;

exports[`Title should match snapshots for each variant without icon 3`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 0,
        "flexDirection": "row",
        "gap": 16,
      },
      undefined,
    ]
  }
  testID="title-container"
>
  <Text
    style={
      {
        "color": "#000",
        "fontSize": 14,
        "fontWeight": "bold",
      }
    }
  >
    Lorem ipsum dolor sit amet
  </Text>
</View>
`;
